poly/deal.go

```diff
@@ -234,10 +234,10 @@ type Promise struct {
  *    secretPair.Suite == longPair.Suite
  *
  * Returns
- *   A newly constructed Promise
+ *   A newly constructed Deal
  */
-func (p *Promise) ConstructPromise(secretPair *config.KeyPair,
-	longPair *config.KeyPair, t, r int, insurers []abstract.Point) *Promise {
+func (p *Deal) ConstructDeal(secretPair *config.KeyPair,
```

Comment:
Isn’t it possible to make these constructors/initialisers more unified? I think we have (mixed from different structures/interfaces):

* Init
* Create
* Construct
* NewDealer

According to

https://golang.org/doc/effective_go.html#composite_literals <https://golang.org/doc/effective_go.html#composite_literals>
and
http://stackoverflow.com/questions/4498998/how-to-initialize-members-in-go-struct <http://stackoverflow.com/questions/4498998/how-to-initialize-members-in-go-struct>

it should all be ```New*``` anyway, no ```Init```, no ```Create``` and no ```Construct```.

What do you think?

> On 12 Oct 2015, at 18:35, Bryan Ford <notifications@github.com> wrote:
> 
> In poly/deal.go <https://github.com/DeDiS/crypto/pull/44#discussion_r41775112>:
> 
> >   */
> > -func (p *Promise) ConstructPromise(secretPair *config.KeyPair,
> > -	longPair *config.KeyPair, t, r int, insurers []abstract.Point) *Promise {
> > +func (p *Deal) ConstructDeal(secretPair *config.KeyPair,
> I think this method should just be called Construct or Create; no need to include Deal in it since it's a Deal method.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/DeDiS/crypto/pull/44/files#r41775112>.
>